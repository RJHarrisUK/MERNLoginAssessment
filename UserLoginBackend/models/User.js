const mongoose = require("mongoose");
const uniqueValidator = require("mongoose-unique-validator");

const Schema = mongoose.Schema;
const UserSchema = new Schema({

  username: {
    type: String,
    required: true
  },
  email: {
    type: String,
    required: true
  },
  password: {
    type: String
  }
});

UserSchema.plugin(uniqueValidator);

const User = mongoose.model('User', UserSchema);
module.exports = UserSchema;
module.exports = User;
