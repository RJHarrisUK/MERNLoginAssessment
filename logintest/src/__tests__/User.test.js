import React from "react";
import User from '../components/User.js';
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router-dom";

it("render a user comp", () => {

    const wrapper = renderer.create(
        <MemoryRouter>

            <User />

        </MemoryRouter>
    ).toJSON();

    expect(wrapper).toMatchSnapshot();

}); 