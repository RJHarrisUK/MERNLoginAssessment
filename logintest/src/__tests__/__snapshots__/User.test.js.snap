// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a user comp 1`] = `
<div>
  <p />
</div>
`;
